cfme/common/__init__.py

```diff
@@ -155,6 +161,141 @@ def get_tags_db(self):
+                     or pass list with tag objects
+        """
+        if isinstance(tags, dict):
+            for category, tag in tags.items():
+                self.add_tag(category=category, tag=tag)
+        elif isinstance(tags, (list, tuple)):
+            for tag in tags:
+                self.add_tag(tag=tag)
+
+    def remove_tag(self, category=None, tag=None, cancel=False, reset=False):
+        """ Remove tag of tested item
+               Args:
+                   category: category(str)
+                   tag: tag(str) or Tag object
+                   cancel: set True to cancel tag deletion
+                   reset: set True to reset tag changes
+               """
+        view = navigate_to(self, 'EditTags')
+        if isinstance(tag, Tag):
+            category = tag.category.display_name
+            tag = tag.display_name
+        try:
+            row = view.tag_table.row(category="{} *".format(category), assigned_value=tag)
+        except Exception:
+            row = view.tag_table.row(category=category, assigned_value=tag)
+        row[0].click()
+        self._tags_action(view, cancel, reset)
+
+    def remove_tags(self, tags):
+        """Remove multiple of tags
+            Args:
+                tags: pass dict with category name as key, and tag as value,
+                     or pass list with tag objects
+        """
+        if isinstance(tags, dict):
+            for category, tag in tags.items():
+                self.remove_tag(category=category, tag=tag)
+        elif isinstance(tags, (list, tuple)):
+            for tag in tags:
+                self.remove_tag(tag=tag)
+
+    def get_tags(self, tenant="My Company Tags"):
+        """ Get list of tags assigned to item
+            Args:
+                tenant: string, tags tenant, default is "My Company Tags"
+            Returns:
+                List of tags in format "Tag_category: Tag_name"
+        """
+        view = navigate_to(self, 'Details')
+        tags = view.tag.read()
+        if tags == 'No {} have been assigned'.format(tenant):
+            return []
+        return filter(None, re.split(r'(.*?):\s*\S+\s?', tags))
+
+    def _tags_action(self, view, cancel, reset):
+        """ Actions on edit tags page
+            Args:
+                view: View to use these actions(tag view)
+                cancel: Set True to cancel all changes, will redirect to details page
+                reset: Set True to reset all changes, edit tag page should be opened
+        """
+        if reset:
+            view.reset_button.click()
+            view.flash.assert_message('All changes have been reset')
+        elif cancel:
+            view.cancel_button.click()
+            view.flash.assert_success_message('Tag Edit was cancelled by the user')
+        else:
+            view.save_button.click()
+            view.flash.assert_success_message('Tag edits were successfully saved')
```

Comment:
I'd like to see one change here. Currently, if the user passes `cancel=True, reset=True`, they will remain on the edit tags page after clicking the reset button and the cancel button is not clicked anymore. Let's change this conditional so that `cancel` is still considered and it will be clicked, if set to `True`.